apps/examples/testcase/le_tc/kernel/itc_libc_pthread.c

```diff
@@ -0,0 +1,322 @@
+* @brief                :This tc tests pthread_rwlock_init and pthread_rwlock_destroy
+* @API's covered        :pthread_rwlock_init,pthread_rwlock_destroy
+* @Preconditions        :none
+* @Postconditions       :none
+* @return               :void
+*/
+
+static void itc_libc_pthread_pthread_rwlock_init_destroy_n_attr_set(void)
+{
+	int status;
+	pthread_rwlock_t rw_lock;
+	pthread_rwlockattr_t attr = 1;
+
+	status = pthread_rwlock_init(&rw_lock, &attr);
+	TC_ASSERT_EQ("pthread_rwlock_init", status, ENOSYS);
+
+
+	TC_SUCCESS_RESULT();
+}
+
+
+
+/**
+* @fn                   :itc_libc_pthread_pthread_rwlock_trywrlock_n_after_destroy
+* @Description          :Passing NULL pthread_rwlock_tryrdlock should not return success
+* @API's covered        :pthread_rwlock_trywrlock
+* @Preconditions        :none
+* @Postconditions       :none
+* @return               :void
+*/
+
+static void itc_libc_pthread_pthread_rwlock_trywrlock_n_after_destroy(void)
+{
+	int status;
+	FAR pthread_rwlock_t rw_lock;
+
+	status = pthread_rwlock_init(&rw_lock, NULL);
+	TC_ASSERT_EQ("pthread_rwlock_init", status, OK);
+
+	status = pthread_rwlock_destroy(&rw_lock);
+	TC_ASSERT_EQ("pthread_rwlock_destroy", status, OK);
+
+	status = pthread_rwlock_tryrdlock(&rw_lock);
+	TC_ASSERT_EQ("pthread_rwlock_tryrdlock", status, ERROR);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @fn                   :itc_libc_pthread_pthread_unlock_n_after_destroy
+* @Description          :Passing NULL pthread_rwlock_unlock should not return success
+* @API's covered        :pthread_rwlock_destroy
+* @Preconditions        :none
+* @Postconditions       :none
+* @return               :void
+*/
+
+static void itc_libc_pthread_pthread_unlock_n_after_destroy(void)
+{
+	int status;
+	FAR pthread_rwlock_t rw_lock;
+
+	status = pthread_rwlock_init(&rw_lock, NULL);
+	TC_ASSERT_EQ("pthread_rwlock_init", status, OK);
+
+	status = pthread_rwlock_wrlock(&rw_lock);
+	TC_ASSERT_EQ("pthread_rwlock_wrlock", status, OK);
+
+	status = pthread_rwlock_destroy(&rw_lock);
+	TC_ASSERT_EQ("pthread_rwlock_destroy", status, OK);
+
+	status = pthread_rwlock_unlock(&rw_lock);
+	TC_ASSERT_EQ("pthread_rwlock_destroy", status, ERROR);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @fn                   :itc_libc_pthread_pthread_rwlock_tryrdlock_n_after_destory
+* @Description          :Passing NULL pthread_rwlock_tryrdlock should not return success
+* @API's covered        :pthread_rwlock_tryrdlock
+* @Preconditions        :none
+* @Postconditions       :none
+* @return               :void
+*/
+
+static void itc_libc_pthread_pthread_rwlock_tryrdlock_n_after_destory(void)
+{
+	int status;
+	FAR pthread_rwlock_t rw_lock;
+
+	status = pthread_rwlock_init(&rw_lock, NULL);
+	TC_ASSERT_EQ("pthread_rwlock_init", status, OK);
+
+	status = pthread_rwlock_destroy(&rw_lock);
+	TC_ASSERT_EQ("pthread_rwlock_destroy", status, OK);
+
+	status = pthread_rwlock_tryrdlock(&rw_lock);
+	TC_ASSERT_EQ("pthread_rwlock_destroy", status, ERROR);
+
```

Comment:
in TC_ASSERT_EQ , it should be "pthread_rwlock_tryrdlock" as first parameter instead of "pthread_rwlock_destroy"